pkg/segment/context/context_test.go

```diff
@@ -4,27 +4,82 @@ import (
 	"context"
 	"reflect"
 	"testing"
+
+	projectv1 "github.com/openshift/api/project/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	ktesting "k8s.io/client-go/testing"
+
+	"github.com/openshift/odo/pkg/occlient"
 )
 
 func TestGetContextProperties(t *testing.T) {
-	key, value := "preferenceKey", "consenttelemetry"
+	ckey, value := "preferenceKey", "consenttelemetry"
 	ctx := NewContext(context.Background())
-	setContextProperty(ctx, key, value)
+	setContextProperty(ctx, ckey, value)
 
 	got := GetContextProperties(ctx)
-	want := map[string]interface{}{key: value}
+	want := map[string]interface{}{ckey: value}
 
 	if !reflect.DeepEqual(got, want) {
 		t.Errorf("want: %q got: %q", want, got)
 	}
 }
 
 func TestSetComponentType(t *testing.T) {
-	key, value := "componentType", "maven"
+	ckey, value := "componentType", "maven"
 	ctx := NewContext(context.Background())
 	SetComponentType(ctx, value)
 
-	if _, contains := GetContextProperties(ctx)[key]; !contains {
+	if _, contains := GetContextProperties(ctx)[ckey]; !contains {
 		t.Errorf("component type was not set.")
 	}
 }
+
+func TestSetClusterType(t *testing.T) {
+	ckey := "clusterType"
+	fakeClient, fakeClientset := occlient.FakeNew()
+	ctx := NewContext(context.Background())
+	cases := []struct {
+		want  string
+		setup func(*occlient.FakeClientset)
+	}{
+		{
+			"openshift3",
+			nil,
+		},
+		{
+			"openshift4",
+			fakeClusterVersion,
+		},
+		{
+			"vanilla-kubernetes",
+			fakeProjects,
+		},
+	}
+	for _, c := range cases {
+		if c.setup != nil {
+			c.setup(fakeClientset)
+		}
+		SetClusterType(ctx, fakeClient)
+		got, _ := GetContextProperties(ctx)[ckey]
+		if got != c.want {
+			t.Errorf("got: %q, want%q", got, c.want)
+		}
+	}
+}
+
+func fakeClusterVersion(fakeClientset *occlient.FakeClientset) {
```

Comment:
you can return a `configv1.ClusterVersion` here when the user `list` for `clusterversion`. if you want to verify what parameters to add here, maybe add a `PrependReactor` for anything called on the fake client, and then print the arguments. That way you would know which calls you need to mock.